### PR TITLE
docs(website): improve context, provider, API and router docs

### PR DESCRIPTION
**Pull request by christianalfoni**

### CODE REVIEW

File: docs/api/signal.md [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+
+export default [
+  someAction,
+  someOtherSequence
+]
+```
+
+
+## Parallel
+A parallel runs all actions right after each other, even if they return a promise.
```
Comment (henri-hulski) [inline docs/api/signal.md]: Hmm I thought they run in parallel.

Comment (Guria) [inline docs/api/signal.md]: javascript can't run 2 functions at the same time :)
Just compare this sentence with sequence description.
Though I agree with you, current description is very confusing.

Comment (henri-hulski) [inline docs/api/signal.md]: Maybe there could be a deeper technical description of the differences between running actions in sequence or in parallel.

Comment (christianalfoni) [inline docs/api/signal.md]: Any suggestions on what to write instead? :) Cause this is exactly what happens, but yeah... people might have presumptions on parallel

Comment (henri-hulski) [inline docs/api/signal.md]: So what happens in sequence?

Comment (christianalfoni) [inline docs/api/signal.md]: If an action returns a promise it will wait before moving on :)

Comment (christianalfoni) [inline docs/api/signal.md]: Updated text now :-)

**Pull request merged**